src/V3LinkInc.cpp

```diff
@@ -0,0 +1,250 @@
+        m_insStmtp = NULL;  // First thing should be new statement
+        iterateAndNextNull(nodep->bodysp());
+        iterateAndNextNull(nodep->incsp());
+        // Done the loop
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+    virtual void visit(AstNodeFor* nodep) VL_OVERRIDE {  // LCOV_EXCL_LINE
+        nodep->v3fatalSrc(
+            "For statements should have been converted to while statements in V3Begin.cpp");
+    }
+    virtual void visit(AstNodeStmt* nodep) VL_OVERRIDE {
+        if (!nodep->isStatement()) {
+            iterateChildren(nodep);
+            return;
+        }
+        m_insMode = IM_BEFORE;
+        m_insStmtp = nodep;
+        iterateChildren(nodep);
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+    void unsupported_visit(AstNode* nodep) {
+        m_unsupportedHere = true;
+        UINFO(9, "Marking unsupported " << nodep << endl);
+        iterateChildren(nodep);
+        m_unsupportedHere = false;
+    }
+    virtual void visit(AstLogAnd* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstLogOr* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstLogEq* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstLogIf* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstCond* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstCondBound* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    void prepost_visit(AstNodeTriop* nodep) {
+        // Check if we are underneath a statement
+        if (!m_insStmtp) {
+            prepost_non_stmt_visit(nodep);
+        } else {
+            prepost_stmt_visit(nodep);
+        }
+    }
+    void prepost_non_stmt_visit(AstNodeTriop* nodep) {
+        iterateChildren(nodep);
+
+        AstNode* lhsp = nodep->lhsp();
+        AstNode* rhsp = nodep->rhsp();
+        AstConst* constp = VN_CAST(nodep->thsp(), Const);
+        UASSERT_OBJ(nodep, constp, "Expecting CONST");
+        AstConst* newconstp = constp->cloneTree(true);
+
+        lhsp->unlinkFrBack();
+        rhsp->unlinkFrBack();
+
+        AstAssign* assignp;
+        if (VN_IS(nodep, PreSub) || VN_IS(nodep, PostSub)) {
+            assignp = new AstAssign(nodep->fileline(), lhsp,
+                                    new AstSub(nodep->fileline(), rhsp, newconstp));
+        } else {
+            assignp = new AstAssign(nodep->fileline(), lhsp,
+                                    new AstAdd(nodep->fileline(), rhsp, newconstp));
+        }
+        nodep->replaceWith(assignp);
+
+        VL_DO_DANGLING(nodep->deleteTree(), nodep);
+    }
+    void prepost_stmt_visit(AstNodeTriop* nodep) {
+        AstNodeVarRef* varrefp;
+        if (m_unsupportedHere || !(varrefp = VN_CAST(nodep->lhsp(), VarRef))) {
+            nodep->v3error("Unsupported: Incrementation in this context.");
+            return;
+        }
```

Comment:
If it doesn't mess things up should keep iterating just to give user additional warnings as appropriate.

```suggestion
            nodep->v3error("Unsupported: Incrementation in this context.");
            iterateChildren(nodep);
            return;
        }
```

Reply:
I moved `iterateChildred` before this check so it is always executed. It looks as though other visitors iterate children as the very first step so it's consistent